tests/Unit/EventSubscriber/HookEventBridgeTest.php

```diff
@@ -28,7 +28,7 @@ public function testCorrectImplementation(): void
 	public function testGetStaticSubscribedEventsReturnsStaticMethods(): void
 	{
 		$expected = [
-			DataFilterEvent::class => 'onDataFilterEvent',
+			NamedEvent::class => 'onNamedEvent',
```

Comment:
Is there a specific reason we can’t or shouldn’t use the same scheme for `HtmlFilterEvent` based on the class name instead of the individual event names?

Reply:
I'm not sure what you mean by "same scheme".

Reply:
Instead of multiple Event name => callable, having a single Event class name => callable entry in the mapping array.

Reply:
fyi: I don't have a ready event system in mind. You see me live developing the architecture. Nearly nothing in this PR is set in stone (with exception of the PSR-14 usage :slightly_smiling_face:).

It is possible to create a single Event class that could handle every event call but that results in an event class that would a) require the listeners (and calling code) to validate the provided data themself or b) require many getter and setter to validate the data inside the event class, but most of them would be unused in a specific event call. I noted that (a) is the case [here](https://github.com/friendica/friendica/pull/14640#discussion_r1899340672) and started the development of the `HtmlFilterEvent`.

The other extrem side would be to create 130+ single event classes, that are made for one specific event call. That would allow us to remove event `name` because the event class would be unique enough. I think that would be exaggerated at the moment.

As a compromise I try to find "event groups" that can share the same event class. Modifying the HTML for the frontend is such an example (see `HtmlFilterEvent`). Another example could be a `SecurityEvent` on login/logout, where context data (e.g. the `uid`) will be provided.

The `NamedEvent` seems to be the smallest possible event class. If this proves to be the case, then I will create an interface for all other event classes from it.